Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -7,31 +10,271 @@ public static class BalanceStats
+                if (biggestLoss > loss)
+                {
+                    biggestLoss = loss;
+                    personWithBiggestLoss = person[0];
+                }
+            }
+
+            return $"{personWithBiggestLoss} lost the most money. {FormatCurrency(biggestLoss)}.";
+        }
+
+        public static string FormatCurrency(float number)
+        {
+            var originalCulture = CultureInfo.CurrentCulture;
+            var culture = CultureInfo.CurrentCulture = CultureInfo.InvariantCulture;
+            var clonedNumbers = (NumberFormatInfo)culture.NumberFormat.Clone();
+            clonedNumbers.CurrencyNegativePattern = 1;
+            var formatedBiggestLoss = string.Format(clonedNumbers, "{0:C0}", number);
+            CultureInfo.CurrentCulture = originalCulture;
+
+            return formatedBiggestLoss;
+        }
+
+        public static float GetLossFor(string personAndBalance)
+        {
+            var balance = ConvertToBalanceArray(personAndBalance);
+            var loss = 0f;
+            for (int i = 0; i < balance.Length - 1; i++)
+            {
+                if (balance[i + 1] - balance[i] < loss)
+                {
+                    loss = balance[i + 1] - balance[i];
+                }
+            }
+            return loss;
+        }
+
+        public static float GetCurrentBalancefor(string personAndBalance)
+        {
+            var balance = ConvertToBalanceArray(personAndBalance);
+            return balance[balance.Length - 1];
         }
 
         /// <summary>
         /// Return name and current money of the richest person.
         /// </summary>
         public static string FindRichestPerson(string[] peopleAndBalances)
         {
-            return "";
+            if (IsArrayNullOrEmpty(peopleAndBalances))
+            {
+                return "N/A.";
+            }
+            var richestPersonsBalance = GetRichestPersonsBalance(peopleAndBalances);
+
+            var richestPeopleNames = GetPeopleNamesWithSameBalance(peopleAndBalances, richestPersonsBalance, "current");
+            var formatedPeopleNames = GetFormatedPeople(richestPeopleNames);
+
+            var areOrIs = (richestPeopleNames.Length > 1) ? "are" : "is";
+            var peopleOrPerson = (richestPeopleNames.Length > 1) ? "people" : "person";
+            return $"{formatedPeopleNames} {areOrIs} the richest {peopleOrPerson}. ¤{richestPersonsBalance}.";
         }
 
+        public static float GetRichestPersonsBalance(string[] peopleAndBalances)
+        {
+            if (peopleAndBalances.Length == 1)
+            {
+                return GetCurrentBalancefor(peopleAndBalances[0]);
+            }
+            var richestPersonsBalance = GetCurrentBalancefor(peopleAndBalances[0]);
+            for (int i = 0; i < peopleAndBalances.Length - 1; i++)
+            {
+          //      richestPersonsBalance = GetCurrentBalancefor(peopleAndBalances[i]);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5012948)